### ISSUE TITLE
client example with file transfer

### ISSUE THREAD

**Issue by Dobiasd** [LibVNC/libvncserver#83]
Hi,

is there an example on how to make a client and run a file transfer somewhere? I did not find anything and was unable to figure out how to implement one. Can you give me a hint?

Tobias

**Comment by weedsacc** [comment]
i need too

**Comment by antenore** [comment]
I don't see the client implementation, but only the server side.

If this is the case we need first to implement the missing code, and obviously patches are welcome.

Some inspiration can be taken form GNOME Remote Desktop and FreeRDP, they both use fuse on the client side (beside the protocol implementation)

**Comment by bk138** [comment]
Right, there's no libvncclient support for UltraVNC- nor TightVNC-1.3- nor TightVNC-2.x file transfer. Yet.

**Comment by bk138** [comment]
re #624

**Comment by Neustradamus** [comment]
To follow this ticket :)

**Comment by marcofortina** [comment]
I opened https://github.com/LibVNC/libvncserver/pull/719 with a LibVNCClient-side implementation for the legacy `rfbFileTransfer` path.

The PR adds:

- a `HandleFileTransfer` callback in `rfbClient`
- a `FileTransferSend()` helper for client applications
- parsing of incoming legacy `rfbFileTransfer` messages
- a client example at `examples/client/filetransfer.c`
- regression tests for the callback and send helper

The example supports remote directory listing and remote file download against the existing server-side `examples/server/filetransfer` example.

I also smoke-tested it locally:

- started `examples/server/filetransfer` with `-permitfiletransfer`
- downloaded a file with the new client example
- verified the downloaded file with `cmp`
- listed the remote directory and saw the expected file names

This PR intentionally targets the legacy UltraVNC/TightVNC-1.3.x `rfbFileTransfer` message path. TightVNC-2.x file transfer https://github.com/LibVNC/libvncserver/issues/624 is a separate protocol extension and should remain a separate feature request/PR.
